source/solvers/heat_transfer.cc

```diff
@@ -23,589 +27,237 @@
 template <int dim>
 void
 HeatTransfer<dim>::assemble_matrix_and_rhs(
-  const Parameters::SimulationControl::TimeSteppingMethod time_stepping_method)
+  const Parameters::SimulationControl::
+    TimeSteppingMethod /*time_stepping_method*/)
 {
-  assemble_system<true>(time_stepping_method);
+  assemble_system_matrix();
+  assemble_system_rhs();
 }
 
 
 template <int dim>
 void
-HeatTransfer<dim>::assemble_rhs(
-  const Parameters::SimulationControl::TimeSteppingMethod time_stepping_method)
+HeatTransfer<dim>::assemble_rhs(const Parameters::SimulationControl::
+                                  TimeSteppingMethod /*time_stepping_method*/)
```

Comment:
With the new merge of the KINSOL solver this interface should be deprecated now. I think you can remove this alltogether and just use the assemble_system_rhs